posts/inside-rust/2023-06-13-Async-WG-June-Update.md

```diff
@@ -0,0 +1,89 @@
+
+## What Phase 1 enables: using Async Functions in Traits
+
+Phase 1 provides enough functionality to make it possible to write traits that contain async functions that are either *always* `Send` or *never* `Send`. This makes it roughly equivalent to what you can do with the `#[async_trait]` macro, but potentially more efficient, since you can avoid boxing when using static dispatch. It is also possible to fill in support for `dyn` dispatch using a macro. We expect that many async trait users in these early phases will wind up using some sort of macros to remove some boilerplate, and are discussing shipping a rust-lang crate that provides these common utilities.
+
+For example, consider the `HealthCheck` trait shown here:
+
+```rust
+trait HealthCheck {
+    async fn check(&mut self);
+}
+
+impl HealthCheck for MyType {
+    async fn check(&mut self) { /* ... */ }
+}
+```
+
+As written, this `HealthCheck` trait is not guaranteed to return `Send` futures. Using explicit `-> impl Future` notation, however, it is possible to make a variant of the trait that *does* guarantee returning `Send` futures; better yet, it's still possible to write `async fn` in the impl:
+
+```rust
+trait SendHealthCheck {
+    fn check(&mut self) -> impl Future<Output = ()> + Send + '_;
+}
+
+impl SendHealthCheck for MyType {
+    async fn check(&mut self) { /* ... */ }
+}
+```
+
+If the async fn is not `Send`, the [compiler would error](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=0948f287cd7fae0c8ddad3a586b698f4).
+
+## What Phase 2 enables: the "send bound" problem
+
+Phase 1 is great, but it doesn't support writing traits that only *sometimes* require `Send` bounds. This is really needed for traits that will be used in highly generic contexts, like the standard library. To support those kinds of cases, we really want a solution to the "send bounds" problem. We currently have an experimental implementation of one proposal, return type notation, and this month a draft RFC was [posted to Zulip](https://rust-lang.zulipchat.com/#narrow/stream/187312-wg-async/topic/associated.20return.20types.20draft.20RFC/near/356796689) and then reviewed in a [lang team design meeting](https://github.com/rust-lang/lang-team/blob/master/design-meeting-minutes/2023-05-24-return-type-notation.md). That conversation turned up a few other ideas that hadn't been considered yet, and we are talking about them before settling on a final direction. The goal is to open an RFC for the final approach, whatever that winds up being, later this month.
+
+## Timeline and Roadmap
+
+To provide an overview of our progress and the expected timeline, here is our updated roadmap for stabilizing RPITIT, async functions in traits, and addressing the "send bounds" problem:
```

Comment:
`RPITIT` appears here for the first time in this post without definition. Above it is always referred to as "`-> impl Future` notation"--which seems like an intuitive, clear description. Below this point it is always `RPITIT`. It would be more understandable if the terminology either stayed consistent, or if using the acronym from this point on is preferred then it should at least be defined (return position impl trait in traits).

Reply:
thanks, fixed.